### PR TITLE
Log nodes running pods in test namespaces

### PR DESCRIPTION
**Pull request by maya-r**
Some of our tests don't run VMs (e.g. guestfs tests) and we are missing valuable info by only recording information from nodes that run virt-launcher pods.

**Special notes for your reviewer**:
Basically, re-doing #6687 with the requested change by roman

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmohr** [comment]
/approve

the approach looks fine to me. Did not review in detail, therefore no lgtm atm.

**Comment by rmohr** [comment]
Note that if you are debugging https://storage.googleapis.com/kubevirt-prow/reports/flakefinder/kubevirt/kubevirt/flakefinder-2022-11-07-168h.html#row0, the node logs will very likely not help you. You can see on https://storage.googleapis.com/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.23-sig-storage/1587462033018720256/artifacts/k8s-reporter/1_overview.log (and all similar failing ones) this line:

```
kubevirt-test-default1   pod/libguestfs-tools-pvc-fs-with-different-uid      1/1     Terminating   0          25s   10.244.196.168   node01   <none>           <none>
```

Very likely the issue is that these serial tests are reusing the pod name `libguestfs-tools-pvc-fs-with-different-uid` and not waiting for it to be gone. The reason for not waiting is unpredictable cleanup time, and it significantly speeds up tests to not wait for the pods to disappear. Right now we tackle it by randomizing pod names like this one (another common approach would be different random namespaces, debateable if we have the best approach, but it is like this right now).

Most flaky tests are not flaky due to node issues and you can see things in the cluster-level collections, but definitely ok to extend the logging.

File: tests/reporter/kubernetes.go [hunk 1]
```diff
@@ -1040,24 +1040,26 @@ func (r *KubernetesReporter) logClusterOverview() {
...
+				nodeMap[pod.Spec.NodeName] = struct{}{}
+			}
 		}
-	}
 
-	nodes := []string{}
-	for k := range nodeMap {
-		nodes = append(nodes, k)
+		for k := range nodeMap {
+			nodes = append(nodes, k)
```
Comment (xpivarc) [inline tests/reporter/kubernetes.go:1061]: `nodes` is not set so you can end up with duplicates. You probably want to make the `nodeMap` out of the scope of the `for` cycle and just transform it on return.

**Review by xpivarc — CHANGES_REQUESTED** [review]

File: tests/reporter/kubernetes.go (hunk 1 above)
Comment (maya-r) [reply tests/reporter/kubernetes.go]: Thanks, fixed.

**Review by xpivarc — APPROVED** [review]

**Pull request merged**